src/jit.c

```diff
@@ -0,0 +1,1566 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+/* This JIT implementation has undergone extensive modifications, heavily
+ * relying on the ubpf_jit_[x86_64|arm64].[ch] from ubpf. The original
+ * ubpf_jit_[x86_64|arm64].[ch] file served as the foundation and source of
+ * inspiration for adapting and tailoring it specifically for this JIT
+ * implementation. Therefore, credit and sincere thanks are extended to ubpf for
+ * their invaluable work.
+ *
+ * Reference:
+ *   https://github.com/iovisor/ubpf/blob/main/vm/ubpf_jit_x86_64.c
+ *   https://github.com/iovisor/ubpf/blob/main/vm/ubpf_jit_arm64.c
+ */
+
+#if !defined(__x86_64__) && !defined(__aarch64__)
+#error "This implementation is dedicated to x64 and arm64."
+#endif
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#if HAVE_MMAP
```

Comment:
No `HAVE_MMAP` check should appear here since we assume BSD/macOS/Linux is used. For Windows support, we may provide a wrapper header `mman.h` instead. Again, in current JIT compiler, no more `mmap` checks.